### DISCUSSION TITLE
UNKNOWN USER committed to my project - 2 commits by unknown user "ich123"

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21603] [New to GitHub]
<p><strong>Summary</strong>:   Someone other than myself seems to be listed as the “committer” of the only 2 commits committed on this project (test lesson tutorial project).</p>
<p>I committed 2 commits (from the command line in Linux) and for some reason the system displays another user as the person that committed the files.    I gave no permissions to anyone else.</p>
<p>Me:  <strong>mike200000000</strong><br>
Username displayed as the committer for some reason:   <strong>ich123</strong></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/e/3e2f4aff89434d9cf702707e1605933dd0b65a1b.png" data-download-href="/uploads/short-url/8S6VwJCubiffk9Vr7IXp4SE6V4L.png?dl=1" title="ich123" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10137/181065789-23ae9e94-3601-434b-8483-0882fb574646.png" alt="ich123" data-base62-sha1="8S6VwJCubiffk9Vr7IXp4SE6V4L" width="690" height="378" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/e/3e2f4aff89434d9cf702707e1605933dd0b65a1b_2_690x378.png, https://aws1.discourse-cdn.com/github/original/3X/3/e/3e2f4aff89434d9cf702707e1605933dd0b65a1b.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/e/3e2f4aff89434d9cf702707e1605933dd0b65a1b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/e/3e2f4aff89434d9cf702707e1605933dd0b65a1b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">ich123</span><span class="informations">974×534 83.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I tried searching online for this specific username thinking it might be some sort of default or something, but I’m not finding anything.</p>
<p>As you can see below, the push seemed to work and my project page only indicates 2 commits (the ones accomplished with the commands below)</p>
<p>Why does “<strong>ich123</strong>”  show up as the user that committed my files?</p>
<p>*** See here there are only 2 commits (correctly showing mike200000000 as the owner)</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/mike200000000/Story/network" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/10137/181065791-26f26784-21de-49db-adb6-e594daa2b348.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d_2_10x10.png"></div>

<h3><a href="https://github.com/mike200000000/Story/network" target="_blank" rel="noopener nofollow ugc">mike200000000/Story</a></h3>

<p><a href="https://github.com/mike200000000/Story/network" target="_blank" rel="noopener nofollow ugc">rk</a></p>

  <p><span class="label1">My masterpiece. Contribute to mike200000000/Story development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>*** <strong>HOWEVER</strong>,  on this page you can clearly see “ich123”  is the committer.<br>
[<strong>You must click the link below</strong> in order to <strong>actually see user “ich123”</strong>; it’s not displaying on this inline part of the post]</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/mike200000000/Story" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/10137/181065793-893076cf-19bd-4ba2-8278-6922a8e8a12e.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/e/5e219cc5f2f17077ece6d5d27d4f78dc5237ea9d_2_10x10.png"></div>

<h3><a href="https://github.com/mike200000000/Story" target="_blank" rel="noopener nofollow ugc">mike200000000/Story</a></h3>

  <p><span class="label1">My masterpiece. Contribute to mike200000000/Story development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>// COMMAND LINE OUTPUT:     git log of what I’m checking in</p>
<pre><code class="lang-auto">&gt; git log
&gt; commit 10dc48ad1a8283bcfa24b2f4fdc1876fbba3e02f (HEAD -&gt; master)
&gt; Author: My name &lt;me@me.com&gt;
&gt; Date:   Sat May 22 21:45:58 2021 -0500
&gt; 
&gt;     complete chapter 2 and 3
&gt; 
&gt; commit bd5f41713cb0b961d240922cfbb9040154c69573
&gt; Author: My name &lt;me@me.com&gt;
&gt; Date:   Sat May 22 20:47:23 2021 -0500
&gt; 
&gt;     Initial Commit - Complete Chapter 1
</code></pre>
<p>// COMMAND LINE OUTPUT:  pushing     (this is the ONLY commit made on GitHub)</p>
<pre><code class="lang-auto">&gt; git push -u origin master
&gt; Username for 'https://github.com': mike200000000
&gt; Password for 'https://mike200000000@github.com': 
&gt; Enumerating objects: 7, done.
&gt; Counting objects: 100% (7/7), done.
&gt; Delta compression using up to 2 threads
&gt; Compressing objects: 100% (6/6), done.
&gt; Writing objects: 100% (7/7), 623 bytes | 311.00 KiB/s, done.
&gt; Total 7 (delta 0), reused 0 (delta 0)
&gt; To https://github.com/mike200000000/Story.git
&gt;  * [new branch]      master -&gt; master
&gt; Branch 'master' set up to track remote branch 'master' from 'origin'.
</code></pre>
<p>Thank you to anyone that can explain what’s going on.   Is this possibly a bug with GitHub ?</p>
<p>Best regards,<br>
Mike</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>GitHub uses the author email in the commits to connect commits to an account, and it looks like that particular address you’re using is already associated with that other account you are seeing.</p>
<p>To fix this, see <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address" class="inline-onebox">Setting your commit email address - GitHub Docs</a> for how to set up your mail address correctly. There’s also an option in case you want to keep your mail address private. Changing past commits requires <a href="https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History">rewriting history</a>, unless you just want to discard the old commits and start over.</p>

**Comment by ghost** [comment]
<p>Thank you very much !   That makes perfect sense.   I did indeed choose an obscure non-existing email for myself on my local machine and that must be the reason.    When I get back to my system I do development on I will implement what you suggested.   I’m pretty certain that’s the answer.  Thank you!</p>
